### PR TITLE
#145: set 'agent', 'item' and 'job' of Script inputs

### PR DESCRIPTION
**Pull request by kovax**

### CODE REVIEW

**Comment by samottitusgt** [comment]
I just noticed this in the logs. At first there is a value for the item then it is null. The fix did work when I validate it.
image

File: kernel/src/main/java/org/cristalise/kernel/scripting/Script.java [hunk 1]
```diff
@@ -564,7 +564,8 @@ public boolean setInputParamValue(String name, Object value, boolean overwrite)
 
             Bindings bindings = context.getBindings(ScriptContext.ENGINE_SCOPE);
 
-            if (overwrite || ! bindings.containsKey(name) || bindings.get(name) == null) {
+            if (       ! bindings.containsKey(name) || bindings.get(name) == null
+                    || ( overwrite && value != null ) ) {
```
Comment (taurus227) [inline kernel/src/main/java/org/cristalise/kernel/scripting/Script.java]: Add unit test to cover all corner cases.

File: restapi/src/main/java/org/cristalise/restapi/ItemRoot.java [hunk 2]
```diff
@@ -558,10 +558,16 @@ public Response getJobFormTemplate(
...
+                CastorHashMap inputs = (CastorHashMap) thisJob.getActProps().clone();
+
+                inputs.put(Script.PARAMETER_AGENT, agent);
+                inputs.put(Script.PARAMETER_ITEM, item);
+                inputs.put(Script.PARAMETER_JOB, thisJob);
 
                 // set outcome if required
                 if (thisJob.hasOutcome()) {
-                    return Response.ok(new OutcomeBuilder(thisJob.getSchema(), false).generateNgDynamicForms(thisJob.getActProps())).build();
+                    return Response.ok(new OutcomeBuilder(thisJob.getSchema(), false).generateNgDynamicForms(inputs)).build();
```
Comment (taurus227) [inline restapi/src/main/java/org/cristalise/restapi/ItemRoot.java]: Add `thisJob.getActProps()` into the `inputs` map to keep backwards-compatibility.

Comment (taurus227) [reply restapi/src/main/java/org/cristalise/restapi/ItemRoot.java]: Ah already done, cool.

File: kernel/src/main/java/org/cristalise/kernel/scripting/Script.java (hunk 1 above)
Comment (taurus227) [reply kernel/src/main/java/org/cristalise/kernel/scripting/Script.java]: Will be done in #147
